cmd/node.go

```diff
@@ -251,7 +296,43 @@ func (node *Node) Start(exitChannels ...*chan struct{}) {
 			}
 		}
 	}
-	node.IsRunning = true
+
+	node.statusMutex.Lock()
+	// Load the node status.
+	// This is identify whether the node is initialized for the first time or it's a restart.
+	status, err := node.LoadStatus()
+	// FIXME: Replace panics with return value,
+	// and leave the decision making to the client library of the core whether to crash the app.
+	if err != nil {
+		panic("Failed to load node status")
+	}
```

Comment:
this needs to be fixed before we can merge this in. don't like having panics lying around